fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -372,25 +375,28 @@ public boolean usesSubspaceKeyCounter() {
 
     @Nonnull
     public Collection<RecordType> recordTypesForIndex(@Nonnull Index index) {
-        if (getUniversalIndexes().contains(index)) {
-            return getRecordTypes().values();
-        }
-        List<RecordType> result = new ArrayList<>();
-        for (RecordType recordType : getRecordTypes().values()) {
-            if (recordType.getIndexes().contains(index)) {
-                return Collections.singletonList(recordType);
-            } else if (recordType.getMultiTypeIndexes().contains(index)) {
-                result.add(recordType);
-            }
-        }
-        for (SyntheticRecordType<?> recordType : getSyntheticRecordTypes().values()) {
-            if (recordType.getIndexes().contains(index)) {
-                return Collections.singletonList(recordType);
-            } else if (recordType.getMultiTypeIndexes().contains(index)) {
-                result.add(recordType);
+        return recordTypesForIndex.computeIfAbsent(index, idx -> {
```

Comment:
Should probably be `MapsUtils.computeIfAbsent` to avoid the lockup bug

Reply:
ah yes...

Reply:
fixing...